sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubClient.cs

```diff
@@ -66,7 +58,25 @@ public class EventHubClient : IAsyncDisposable
         ///   Event Hub will result in a connection string that contains the path.
         /// </remarks>
         ///
-        public EventHubClient(string connectionString) : this(connectionString, null)
+        public EventHubClient(string connectionString) : this(connectionString, null, null)
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to use for connecting to the Event Hubs namespace; it is expected that the shared key properties are contained in this connection string, but not the Event Hub path.</param>
+        /// <param name="eventHubPath">The path of the specific Event Hub to connect the client to.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Event Hub itself, it will contain the path to the desired Event Hub,
+        ///   and can be used directly without passing the <paramref name="eventHubPath" />.  The path to the Event Hub should be
+        ///   passed only once, either as part of the connection string or separately.
+        /// </remarks>
+        ///
+        public EventHubClient(string connectionString,
```

Comment:
We originally had this constructor too. And I thought that it was decided to remove it during that last API review, because there was ambiguity about which EventHub to use if the connection string contained "EntityPath" too. Do we use the one from the connection string? Or the one they explicitly passed in? Or do we throw an exception because it may be a misconfiguration?

Reply:
Heh, I recall the conversation the other way.   The board requested to add it because of the "bad decisions" around the connection string in the portal.    I validate (and document explicitly) that you cannot pass the event hub in both places;  things go boom if you try.

Since the connection string in the portal that is most visible and accessible is the namespace-level one, I think this will be the primary use case, honestly.  I don't see folks digging into the portal to get to the exact Event Hub (or knowing to do so).

So why not use the expanded constructor and pass in a shared key credential?   The actual host for the namespace only appears in the portal in the connection string, from what I can see.  Consumers would have to know to take their Event Hubs instance name and prepend it to the magic url value.   That also doesn't seem to be the easy path.... so, here we are with a few more constructors.

Reply:
Fair enough. I'll add it back. I think it's a common use-case.

Reply:
> Since the connection string in the portal that is most visible and accessible is the namespace-level one, I think this will be the primary use case, honestly. I don't see folks digging into the portal to get to the exact Event Hub (or knowing to do so).

+100 for this